tests/migration_test.go

```diff
@@ -1438,6 +1437,79 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
 				})
 			})
 		})
+
+		Context("with restricted CPU models", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			prepareMigration := func(cpuModel string) *v1.VirtualMachineInstanceMigration {
+				vmi.Spec.Domain.CPU = &v1.CPU{
+					Model: cpuModel,
+				}
+				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\necho 'hello'\n")
+
+				By("Starting the VirtualMachineInstance")
+				vmi = runVMIAndExpectLaunch(vmi, 240)
+
+				By("Checking that the VirtualMachineInstance console has expected output")
+				Expect(console.LoginToCirros(vmi)).To(Succeed())
+
+				// execute a migration, wait for finalized state
+				migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+
+				return migration
+			}
+
+			disposeVMI := func(vmi *v1.VirtualMachineInstance) {
+				// delete VMI
+				By("Deleting the VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+				By("Waiting for VMI to disappear")
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 240)
+			}
+
+			executeTest := func(cpuModel, featureName string, enable, positive bool) {
+				if enable {
+					tests.EnableFeatureGate(featureName)
+				} else {
+					tests.DisableFeatureGate(featureName)
+				}
+
+				By("Starting a Migration")
+				migration := prepareMigration(cpuModel)
+				migration, err = virtClient.VirtualMachineInstanceMigration(migration.Namespace).Create(migration)
+				if positive {
+					Expect(err).NotTo(HaveOccurred())
+				} else {
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("CPUNotMigratable"))
+				}
+			}
+
+			BeforeEach(func() {
+				vmi = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+			})
+
+			AfterEach(func() {
+				disposeVMI(vmi)
+			})
+
+			table.DescribeTable("should block migration for restricted CPU modes",
+				func(cpuModel, featureName string, enable, positive bool) {
+					executeTest(cpuModel, featureName, enable, positive)
+				},
+				table.Entry("with host-model cpu", "host-model", "MigratableHostModel", false, false),
+				table.Entry("with cpu-passthrough", "host-passthrough", "MigratableHostPassthrough", false, false),
```

Comment:
I guess s/MigratableHostModel/WithHostModelCPU and s/MigratableHostPassthrough/WithHostPassthroughCPU ?